### ISSUE TITLE
Is PidCache ever cleaned if remote node is brutally killed ?

### ISSUE THREAD

**Issue by jrouaix** [AsynkronIT/protoactor-dotnet#458]
My guess is no :

- I have a consul cluster
- and multiple nodes talking to the same virtual actor.
- I kill the node on which the virtual actor is running
* SOMETIMES (yes it's too easy when it's always)
- then I run in an endless loop of split brains (meaning i.m.o) distinct "clients" talk to distinct instances of this actors

Will provide more details here as I find some ...

**Comment by jrouaix** [comment]
ok now I saw the `EndpointTerminatedEvent` or `RemoteTerminate` handled in `EndpointWatcher` that throw a bunch of `Terminated`. I assume it's for the `PidCacheWatcher` previously subscribed to be notified.

How likely it's possible that the endpoint is terminated **before** all messages passing system gets `PidCacheWatcher` properly subscribed ?

**Comment by jrouaix** [comment]
Ho, also `PidCache` is subscribed to `MemberStatusEvent` now how likely the remote member is detected down (and  `MemberStatusEvent` just between `ActorPidResponse` arrive and Pid is add to the cache ?
https://github.com/AsynkronIT/protoactor-dotnet/blob/361b4cd0ea19f2b0e2e2717e7b5dfc4cde692bf8/src/Proto.Cluster/Cluster.cs#L93-L101

**Comment by rogeralsing** [comment]
Oh, race conditions... 
TLDR; you get multiple activations of the same virtual actor on different nodes?
And some clients never release their association with the incorrect activation?

**Comment by rogeralsing** [comment]
Pid cache has been removed. 
We are adding a new design for this on `Cluster.RequestAsync`
